x-pack/plugins/fleet/common/errors.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+/* eslint-disable max-classes-per-file */
+
+import type { FleetErrorType } from './types';
+
+export class IngestManagerError extends Error {
```

Comment:
it would be great to rename this error to FleetError, though that might touch many files

Reply:
I will try to come with a separate PR for that (just after that one to avoid conflict) but with typescript this should be pretty easy to rename :)